multichain-testing/tools/agd-lib.js

```diff
@@ -0,0 +1,176 @@
+// @ts-check
+import assert from 'node:assert';
+
+const { freeze } = Object;
+
+const agdBinary = 'kubectl';
+const binaryArgs = [
+  'exec',
+  '-i',
+  'agoriclocal-genesis-0',
+  '-c',
+  'validator',
+  '--tty=false',
+  '--',
+  'agd',
+];
+
+/**
+ * @param {Record<string, string | undefined>} record - e.g. { color: 'blue' }
+ * @returns {string[]} - e.g. ['--color', 'blue']
+ */
+export const flags = (record) => {
+  // TODO? support --yes with boolean?
+
+  /** @type {[string, string][]} */
+  // @ts-expect-error undefined is filtered out
+  const skipUndef = Object.entries(record).filter(([_k, v]) => v !== undefined);
+  return skipUndef.map(([k, v]) => [`--${k}`, v]).flat();
+};
+
+/**
+ * @callback ExecSync
+ * @param {string} file
+ * @param {string[]} args
+ * @param {{ encoding: 'utf-8' } & { [k: string]: unknown }} opts
+ * @returns {string}
+ */
+
+/**
+ * @param {{ execFileSync: ExecSync }} io
+ */
+export const makeAgd = ({ execFileSync }) => {
```

Comment:
I think we'll want to DRY this with other helpers and I'd also like to simplify it. but this works for now